### PR TITLE
feat(styles): L3-3261 updated new tokens

### PR DESCRIPTION
**Pull request by constantinehuzenko**
**Jira ticket**

[L3-3261](https://phillipsauctions.atlassian.net/browse/L3-3261)

**Screenshots**
| Before | After |
| ------ | ----- |
| image | image |

**Summary**

updated tokens for spacing, text and headers

**Change List (describe the changes made to the files)**

- feat(_vars): added new variables and line-height for existing variables
- feat(types.ts): added new text type - string
- feat(_type.scss): added new text types

**Acceptance Test (how to verify the PR)**

- Use [link](https://figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=10247-38112&m=dev) to verify new type tokens
- Use [link](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=6591-3391&m=dev) to verify new space tokens

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- Post logs, screenshots, etc

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-3261]: https://phillipsauctions.atlassian.net/browse/L3-3261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by scottdickerson — COMMENTED** [review]
The badge, label, and button `letter-spacing` and `line height` seem off in `_types.scss`

**Comment by scottdickerson** [comment]
Font Size and Line height seem incorrect for $header1 
image

**Comment by scottdickerson** [comment]
we should specify the updates to `line-height` too for the different breakpoints text: 
image

**Comment by scottdickerson** [comment]
type tokens page needs to be udpated to have a String type token page: 
image

**Comment by scottdickerson** [comment]
the body line-heights don't seem correct: 
image

image

**Review by scottdickerson — CHANGES_REQUESTED** [review]
Seeing lots of discrepancies still between the Figma and our tokens story

File: src/components/Link/_link.scss [hunk 1]
```diff
@@ -99,4 +99,62 @@
...
+    letter-spacing: 1px;
+    line-height: $link-label-line-height;
+    text-decoration: underline;
+
+    &:hover {
+      border-bottom: 0;
+    }
+  }
+
+  &--snwHeaderLink {
```
Comment (adietrich3074) [inline src/components/Link/_link.scss:116]: This looks like a duplicate of line 128. The styles are slightly different under 128 though so maybe consolidate them and remove the duplicate styles. Let me know if this was intentional and I'll review the PR again

**Comment by scottdickerson** [comment]
I'm not sure what is causing these warnings but we should clean them up before delivering 
```
eprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ╷
100 │       --#{$name}-size#{$i}: #{$font-start - ($i - 1) * $step}rem;
    │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
... │
106 │ ┌   @media (max-width: 960px) {
107 │ │     --heading-size0: 2.59rem;
108 │ │     --quote-size: 1.5rem;
109 │ │     --quote-line-height: 2rem;
110 │ │ 
111 │ │     @include font-size(4, 'heading', 1.5, 1.75);
112 │ │     @include font-size(3, 'string', 1, 1.25);
113 │ │     @include font-size(3, 'body', 1, 1.5);
114 │ │ 
115 │ │     --label-size1: 1rem;
116 │ │     --label-size2: 0.75rem;
117 │ │     --label-size3: 0.56rem;
118 │ │     --link-button-label-size: 0.75rem;
119 │ │     --link-label-line-height: 0.75rem;
120 │ │     --button-label-line-height: 1rem;
121 │ │     --badge-label-size: 0.5rem;
122 │ │     --badge-label-line-height: 0.67rem;
123 │ │     --snw-header-link-size: 1rem;
124 │ │     --snw-header-link-line-height: 1.5rem;
125 │ │     --snw-flyout-link-size: 1rem;
126 │ │     --snw-flyout-link-line-height: 1.75rem;
127 │ │   }
    │ └─── nested rule
    ╵
    src/scss/_vars.scss 100:5               font-size()
    src/scss/_vars.scss 133:3               @use
    src/scss/_sharedClasses.scss 1:1        @forward
    src/scss/_allPartials.scss 1:1          @use
    src/components/Button/_button.scss 1:1  @use
    src/componentStyles.scss 7:1            root stylesheet

Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ╷
101 │       --#{$name}-line-height-size#{$i}: #{$line-start - ($i - 1) * $step}rem;
    │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
... │
106 │ ┌   @media (max-width: 960px) {
107 │ │     --heading-size0: 2.59rem;
108 │ │     --quote-size: 1.5rem;
109 │ │     --quote-line-height: 2rem;
```

File: src/components/Text/types.ts [hunk 2]
```diff
@@ -11,6 +11,9 @@ export enum TextVariants {
   body1 = 'body1',
   body2 = 'body2',
   body3 = 'body3',
+  string1 = 'string1',
```
Comment (adietrich3074) [inline src/components/Text/types.ts:14]: The naming is confusing here. What is the use case for a string in this context?

File: src/scss/_vars.scss [hunk 3]
```diff
@@ -145,14 +180,29 @@ $display-size4: var(--heading-size4);
 
 // HEADING
 $heading-size1: var(--heading-size1);
+$heading-line-height-size1: var(--heading-line-height-size1);
```
Comment (adietrich3074) [inline src/scss/_vars.scss:183]: Is there a reason we're adding line height tokens? Wouldn't heading1 always have the same line height associated with it? Do we have a need to mix and match tokens?

File: src/scss/_vars.scss [hunk 4]
```diff
@@ -161,35 +211,44 @@ $email-label-size: var(--label-size2);
 $cta-lg-label-size: var(--label-size1);
 $text-label-size: var(--label-size3);
 $badge-label-size: var(--label-size3);
+$link-button-label-size: var(--link-button-label-size);
```
Comment (adietrich3074) [inline src/scss/_vars.scss:214]: Same question as the heading line height tokens.

File: src/scss/_vars.scss [hunk 5]
```diff
@@ -95,44 +95,79 @@ $text-tokens: $button, $cta-sm, $cta-lg, $email, $label, $badge, $display0, $dis
...
-  --heading-size2: 1.95rem;
-  --heading-size3: 1.56rem;
-  --heading-size4: 1.56rem;
-  --body-size1: 1.25rem;
-  --body-size2: 1rem;
-  --body-size3: 0.75rem;
-  --label-size1: 1.25rem;
-  --label-size2: 1rem;
-  --label-size3: 0.75rem;
+@mixin font-size($to, $name, $font-start, $line-start, $step: 0.25) {
```
Comment (adietrich3074) [inline src/scss/_vars.scss:98]: I like that you made this dynamically generated. Much easier to add new tokens if needed.

**Review by adietrich3074 — CHANGES_REQUESTED** [review]
Aside from the duplicate styles comment the rest of this PR looks great! The other comments are just looking for clarification and aren't blockers

File: src/components/Text/types.ts (hunk 2 above)
Comment (constantinehuzenko) [reply src/components/Text/types.ts:14]: tbh I didn't ask them, this is what design provided, looks like it's gonna be used for Inputs and etc

File: src/scss/_vars.scss (hunk 3 above)
Comment (constantinehuzenko) [reply src/scss/_vars.scss:183]: we had the same line height before, and now we have different for each case. so we have to do that

File: src/scss/_vars.scss (hunk 4 above)
Comment (constantinehuzenko) [reply src/scss/_vars.scss:214]: not sure what you mean here

**Review by adietrich3074 — APPROVED** [review]
Changes look good to me 👍

**Review by scottdickerson — APPROVED** [review]
LGTM!

**Pull request merged**